### ISSUE TITLE
Error in function create_json: ValueError('need at least one array to concatenate',) 

### ISSUE THREAD

**Issue by nyxrobotics** [pieterblok/maskal#6]
I am new to deep learning.
I took some images to create a dataset for yolact++.
I read the README and understood how to set the path, but I didn't understand how to use auto_annotate.py.
For example, if my image has objects that can be recognized by resnet101_reducedfc.pth (I especially want to recognize a cup), is it possible to automatically create polygons for labelme?
Also, I would like to know how to create network_config.

**Comment by nyxrobotics** [comment]
I was able to get the pretrain behavior using my_maskAL.yaml.
I changed folders and classes and minority_classes.

```xml
# Parameters for maskAL
# python maskAL.py --config maskAL.yaml
# explanation can be found here: https://github.com/pieterblok/maskal

# folders
weightsroot: "./noodle/weights"
resultsroot: "./noodle/results"
dataroot: "./noodle/datasets"
initial_train_dir: "./noodle/datasets/initial_train"
traindir: "./noodle/datasets/train"
valdir: "./noodle/datasets/val"
testdir: "./noodle/datasets/test"

# data options
use_initial_train_dir: False
duplicate_initial_model_and_data: False
initial_train_file: ""

# network parameters
network_config: COCO-InstanceSegmentation/mask_rcnn_X_101_32x8d_FPN_3x.yaml
pretrained_weights: COCO-InstanceSegmentation/mask_rcnn_X_101_32x8d_FPN_3x.yaml

# training-parameters
cuda_visible_devices: '0'
classes: ['curry_noodles', 'seafood_noodles', 'soy_sauce_noodles']
transfer_learning_on_previous_models: True
learning_rate: 0.01
warmup_iterations: 500
train_iterations_base: 2500
train_iterations_step_size: 2500
step_image_number: 500
eval_period: 500
checkpoint_period: -1
weight_decay: 0.0001
learning_policy: 'steps_with_decay'
step_ratios: [0.5, 0.8]
gamma: 0.1
train_batch_size: 1
num_workers: 2

# train-sampler
train_sampler: "TrainingSampler"
minority_classes: ['curry_noodles', 'seafood_noodles', 'soy_sauce_noodles']
repeat_factor_smallest_class: 1.0

# evaluation-parameters
confidence_threshold: 0.5
nms_threshold: 0.2

# active-learning sampling
experiment_name: 'exp1'
strategy: 'uncertainty'
mode: 'mean'
initial_datasize: 5
pool_size: 50
equal_pool_size: True
loops: 5
dropout_probability: 0.25
mcd_iterations: 10
iou_thres: 0.5
auto_annotate: True
export_format: labelme
supervisely_meta_json: "./datasets/meta.json"
```

After that, I annotated inside the annotate folder with labelme.
But I got this error.
```bash
train.json
  0%|               
Cannot find the class name (please check the annotation files)
  5%|████████▊
2022-12-23 20:19:23,356 - active_learning.sampling.prepare_dataset - ERROR - Error in function create_json: ValueError('need at least one array to concatenate',) 
2022-12-23 20:19:23,356 - active_learning.sampling.prepare_dataset - ERROR - Cannot create initial-datasets 
```

What does this error mean?

**Comment by pieterblok** [comment]
> I was able to get the pretrain behavior using my_maskAL.yaml. I changed folders and classes and minority_classes.
> 
> ```
> # Parameters for maskAL
> # python maskAL.py --config maskAL.yaml
> # explanation can be found here: https://github.com/pieterblok/maskal
> 
> # folders
> weightsroot: "./noodle/weights"
> resultsroot: "./noodle/results"
> dataroot: "./noodle/datasets"
> initial_train_dir: "./noodle/datasets/initial_train"
> traindir: "./noodle/datasets/train"
> valdir: "./noodle/datasets/val"
> testdir: "./noodle/datasets/test"
> 
> # data options
> use_initial_train_dir: False
> duplicate_initial_model_and_data: False
> initial_train_file: ""
> 
> # network parameters
> network_config: COCO-InstanceSegmentation/mask_rcnn_X_101_32x8d_FPN_3x.yaml
> pretrained_weights: COCO-InstanceSegmentation/mask_rcnn_X_101_32x8d_FPN_3x.yaml
> 
> # training-parameters
> cuda_visible_devices: '0'
> classes: ['curry_noodles', 'seafood_noodles', 'soy_sauce_noodles']
> transfer_learning_on_previous_models: True
> learning_rate: 0.01
> warmup_iterations: 500
> train_iterations_base: 2500
> train_iterations_step_size: 2500
> step_image_number: 500
> eval_period: 500
> checkpoint_period: -1
> weight_decay: 0.0001
> learning_policy: 'steps_with_decay'
> step_ratios: [0.5, 0.8]
> gamma: 0.1
> train_batch_size: 1
> num_workers: 2
> 
> # train-sampler
> train_sampler: "TrainingSampler"
> minority_classes: ['curry_noodles', 'seafood_noodles', 'soy_sauce_noodles']
> repeat_factor_smallest_class: 1.0
> 
> # evaluation-parameters
> confidence_threshold: 0.5
> nms_threshold: 0.2
> 
> # active-learning sampling
> experiment_name: 'exp1'
> strategy: 'uncertainty'
> mode: 'mean'
> initial_datasize: 5
> pool_size: 50
> equal_pool_size: True
> loops: 5
> dropout_probability: 0.25
> mcd_iterations: 10
> iou_thres: 0.5
> auto_annotate: True
> export_format: labelme
> supervisely_meta_json: "./datasets/meta.json"
> ```
> 
> After that, I annotated inside the annotate folder with labelme. But I got this error.
> 
> ```shell
> train.json
>   0%|               
> Cannot find the class name (please check the annotation files)
>   5%|████████▊
> 2022-12-23 20:19:23,356 - active_learning.sampling.prepare_dataset - ERROR - Error in function create_json: ValueError('need at least one array to concatenate',) 
> 2022-12-23 20:19:23,356 - active_learning.sampling.prepare_dataset - ERROR - Cannot create initial-datasets 
> ```
> 
> What does this error mean?

@nyxrobotics it seems that one of your images is missing an annotation or that one of your annotated instances has a different name. You specified three classes: 'curry_noodles', 'seafood_noodles', 'soy_sauce_noodles'. Maybe one instance in one image has a class with a typo in it. Please double check your annotations to check for these potential errors.

**Comment by nyxrobotics** [comment]
Annotated non-noodle polygons that were automatically created by pretrained_weights remained.
Removing this fixed it. thank you.

**Comment by nyxrobotics** [comment]
Is there a way to reduce the active learning resolution to save GPU memory while the dataset resolution is still high?

**Comment by nyxrobotics** [comment]
@pieterblok Can you tell me the difference between ```classes``` and ```minority_classes``` ?

**Comment by pieterblok** [comment]
> @pieterblok Can you tell me the difference between ```classes``` and ```minority_classes``` ?

@nyxrobotics "classes" is a list of the classes that need to be trained by Mask R-CNN. "minority_classes" are only used in combination with "RepeatFactorTrainingSampler" to battle class-imbalanced datasets, please refer to this: detectron2/data/samplers/distributed_sampler.py (line 129)
